databroker/src/broker.rs

```diff
@@ -1882,6 +1885,71 @@ impl<'a, 'b> AuthorizedAccess<'a, 'b> {
             )),
         }
     }
+
+    async fn validate_actuator_value(
+        &self,
+        vss_id: &i32,
+        data_value: &DataValue,
+    ) -> Result<(), (ActuationError, String)> {
+        let result_entry = self.get_entry_by_id(*vss_id).await;
+        match result_entry {
+            Ok(entry) => {
+                let validation = entry.validate_value(data_value);
+                let vss_path = entry.metadata.path;
+                match validation {
+                    Ok(_) => Ok(()),
+                    Err(UpdateError::NotFound) => {
+                        let message = format!("Could not resolve vss_path {}", vss_path);
+                        Err((ActuationError::NotFound, message))
+                    }
```

Comment:
Might be redundant. 
NotFound is thrown by get_entry_by_id which we use in the method itself but also here in line 1894 for the vss_id lookup. Since we are already in the OK case even a consecutive lookup of the id should not lead to NotFound.

Reply:
`PermissionDenied` and `PermissionExpired` also are redundant, I left to make Rust happy but it just maybe enough if we leave `UnsupportedType`, `WrongType` and `OutOfBounds`

Reply:
Unfortunately I'm not totally satisfied with the current solution :( Now as a default case we cause a panic, which is - in our controlled scenario - totally fine. However, due to the use of the default branch we will not see when a new UpdateError case is added and therefore create a scenario in which a panic is thrown for the newly added UpdateError during conversion to the ActuationError because it was forgotten to be added here. 

Please find a better solution for the default case - in which we don't panic - or change it back to an exhaustive match (with tolerated redundancy - maybe add a comment to show, that the redundancy is accepted). Using an exhaustive match, will create compiler errors in case of an UpdateError being added and not adding a new case to the match - therefore the developer easily sees, that he needs to add it here. 

Sorry for the back and forth!

Reply:
I will leave the extended version with a comment, maybe the `validate_actuator_value` method could be use somewhere else or even at some point `get_entry_by_id` also modified, who knows, it does not hearth to have errors I think.